engine/sorted_collection/skiplist.cpp

```diff
@@ -266,25 +267,17 @@ LockTable::GuardType Skiplist::lockRecordPosition(const DLRecord* record,
                                                   PMEMAllocator* pmem_allocator,
                                                   LockTable* lock_table) {
   while (1) {
-    PMemOffsetType record_offset = pmem_allocator->addr2offset_checked(record);
     PMemOffsetType prev_offset = record->prev;
     PMemOffsetType next_offset = record->next;
     DLRecord* prev = pmem_allocator->offset2addr_checked<DLRecord>(prev_offset);
-    DLRecord* next = pmem_allocator->offset2addr<DLRecord>(next_offset);
 
     auto guard = lock_table->MultiGuard({recordHash(prev), recordHash(record)});
 
     // Check if the list has changed before we successfully acquire lock.
-    if (record->prev != prev_offset || prev->next != record_offset ||
-        record->next != next_offset || next->prev != record_offset) {
+    if (record->prev != prev_offset || record->next != next_offset) {
```

Comment:
Actually you can just check for prev, since you only locks prev and this node. next can no longer be modified if we have acquired lock.

Reply:
it can be modified before we successfully acquire the locks

Reply:
We don't need to care about next, because this lock is used in insertion and replacement. The next node may have been modified, but we don't need to worry. We only need to ensure we successfully lock the prev and this node, and no other node have been inserted between prev and this or the prev has been deleted.